test/core.jl

```diff
@@ -4678,15 +4678,33 @@ end
 @test f14893() == 14893
 @test M14893.f14893() == 14893
 
-# issue #18725
-@test_nowarn @eval Main begin
-    f18725(x) = 1
-    f18725(x) = 2
-end
-@test Main.f18725(0) == 2
-@test_warn "WARNING: Method definition f18725(Any) in module Module18725" @eval Main module Module18725
-    f18725(x) = 1
-    f18725(x) = 2
+let exename = `$(Base.julia_cmd()) --startup-file=no -i`
```

Comment:
this should maybe loop over both with `-i` and without `-i` and test for each intended behavior